File: solutions/cpp/isogram/1/isogram.cpp
#include "isogram.h"
using namespace std;
namespace isogram {
    bool is_isogram(const char * word) {
        int cc[26] = { }; 
        for (int i=0; i<(int)strlen(word); i++){
            char ch=tolower(word[i]);
            if (ch>='a' && ch<='z'){
                if (cc[ch-'a']==1) return false;
                cc[ch-'a'] = 1;
            }
        }
        return true;
    }
}  // namespace isogram

File: solutions/csharp/isogram/1/Isogram.cs
using System;
using System.Collections.Generic;

public static class Isogram
{
    public static bool IsIsogram(string word)
    {
        Dictionary<char, int> cc = new Dictionary<char, int>();
        word = word.ToLower();
        foreach (char ch in word) {
            if (ch != ' ' && ch != '-') {
                if (cc.ContainsKey(ch)) {
                    return false;
                } else {
                    cc[ch] = 1;
                }
            }
        }
        return true;
    }
}
